### PR TITLE
Minor Changes to Tesla Mechanics

### PR DESCRIPTION
**Pull request by Asdjky**
Tesla Miniball maximum now properly enforced.
Miniball power output no longer tied to zap range.
Minimum zap range increased to 2.
Grounding Rod bonus range increased by 1.
Tesla Coils no longer cause zaps to jump.
Coil cooldown restrictions actually work now.
Research/money gain adjusted to compensate for cooldown restrictions.

# General Documentation

### Intent of your Pull Request

This makes some of the Tesla's mechanics less weird, more predictable, and actually impactful. Currently you are extremely unlikely to ever see the Tesla's power output be above 1.7mW (unless you upgrade the Tesla Coils for a straight 100%/200%/etc bonus). This is because of some weird mechanics and decisions on how power generation is calculated. The changes in this PR will mean a roughly 50% increase in power output for the Tesla which will put it a bit under what a default TEG generates.
I've tested these changes and with the default setup power output is always at a minimum of 1.7mW with spikes up to 2.5mW and 3.4mW.

### Why is this change good for the game?
The Tesla's mechanics are easier to reason about and opportunities for experimenting with different Tesla setups are now possible.

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms.
The Tesla's mechanics are now more consistent and impactful. Miniball power generation has been increased because it is no longer tied to zap range. Tesla Coils that are zapped no longer cause a jump because that behavior was putting every coil onto cooldown with a single zap. Tesla Coils now go on cooldown when zapped and can't be zapped again until a short duration (8 seconds, decreased by better parts).

### What should players be aware of when it comes to the changes your PR is implementing?
Casual players who are just interested in having power be setup don't need to be concerned with this change. A default Tesla setup will still work perfectly fine for the station. However, enterprising players will be able to make changes to the Tesla's default setup and see improved results to power output.

### What general grouping does this PR fall under? 
Engineering rebalancing.

### Are there any aspects of the PR that you would like us not to mention on the Wiki?
No

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here.
Grounding Rods now work 1 tile further than they previously did making the default Tesla setup completely safe.
Standard Tesla Coils now generate the full amount of power per zap instead of anywhere from 1-50% of full power. This results in a roughly 50% increase in power output overall. Research Tesla Coils still incur a 95% power loss in exchange for increased research generation.
Tesla Coil cooldowns are now enforced. This means that when a coil is zapped it goes on cooldown and cannot be zapped before the cooldown is up. Level 1 coils have a cooldown of 8 seconds. Coils can be upgraded with capacitors to reduce the cooldown by 2 seconds per capacitor level.

# Changelog
Tesla mechanics tweaked to be more consistent, resulting in a roughly 50% increase in power output and more setup variety.

:cl:  
bugfix: Tesla Miniball maximum now properly enforced  
bugfix: Tesla Coil cooldowns now actually work
tweak: Miniball power output no longer tied to zap range
tweak: Minimum Miniball zap range increased to 2 (was 1)
tweak: Grounding Rod bonus range increase to 3 (was 2)
tweak: Tesla Coils no longer cause zaps to jump
/:cl:

### CODE REVIEW

**Comment by Cartlord** [comment]
I don't know what the hell you did, but the tesla - fully set up - keeps blowing holes in the PA room on Box.

**Comment by Asdjky** [comment]
> 
> 
> I don't know what the hell you did, but the tesla - fully set up - keeps blowing holes in the PA room on Box.

Yea I know what's happening here. It's a combination of me messing up the range check for grounding rods (which was easily fixed) and the cooldowns for tesla coils now properly working. Now that cooldowns actually work there are two positions the Tesla can be in where there are no valid targets in range except the station itself. I'll think about how I should resolve this, I really don't want to do any map changes though.

**Comment by adamsong** [comment]
This has caused the tesla to blow holes in engineering for multiple rounds now. This needs either fixed or unmerged until it is.

**Comment by Asdjky** [comment]
> 
> 
> This has caused the tesla to blow holes in engineering for multiple rounds now. This needs either fixed or unmerged until it is.

I didn't realize this was testmerged, I interpreted Cartlord's comment in a different way. I'll get it fixed right away. Sorry about that.

**Comment by JamieD1** [comment]
Hows the test merges gone on this?

**Comment by Asdjky** [comment]
> 
> 
> Hows the test merges gone on this?

The first few days were a bit rough since I broke it, but after that everything looks good. I haven't personally seen anybody do non-default setups but everything with the default setup looks good in terms of power generation and coil mechanics working as intended.

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 92e694acd1b4ddb5bdab7f42dbf75d255058fcd2
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: da56d524b85eeabf513b808c3ed6a8c9df605f9a

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 6eac6b8678728327d8878fa07e5c0f4655a8fa3e
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 92b18435786db50501d15b57a7d0f399158dcce7

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 6eac6b8678728327d8878fa07e5c0f4655a8fa3e
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 4cf045dcc80de3751ed501a3fdf71a161b4b6078

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 09dc7735c329443b1d7140a2d32a9b7e5388298f
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 8ce5531e4aa90f730f2881ca15125742069cc3b7

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 50e22d907e88fc4aed08712ae4939d0ca006b9bd
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: cfbef971e39f756f57c6351ccba16176fb7ca9b7

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: d244ed8af6b4625f69aff980ca6f669c8eab96cc
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: c9dd238bda1b13104771876231c7f061360f117a

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: b34fc836328b6ea784801f980d65536c09daad58
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 99a8136a2c3ea2386b7417cadce313eef7205037

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 51d15f34c48d85520d7724f0b4ab5cd650e8be05
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 08a5b01226a862a638855658a1fcb349f2180bdf

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: faf404bfec32c473db5ed3f6fa8061fcaa8b01d8
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 261ac2b415670d9615de140a821f0aed25f45412

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: dfdf9d25fba679e22d2ccaca56dc3ff10960e5af
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: a57b89a5cefc62a3059460567fc50d2b23efa5f5

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: dfdf9d25fba679e22d2ccaca56dc3ff10960e5af
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: d723368c137b9977e75b3e26dfb321c788297cdd

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 24d1f239ea0aa6c88caabde6243b7528fe97fb4c
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 1c84e2f8d3d922d24bcc92d7edb9ce9e32d5252e

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: e581b726a44943425eb38bb248e800e792d0fd9d
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: fc7122cf9f35688fd96d1b459cc1c9b61682e752

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 1befeb33968cd7fcd11341f081e90e149979098b
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 87f7ce98fbe6e2119e87704ba716dc7213e89bd9

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: bfcf8bfb1e922f1d6c8f6db19948a76bbfa39116
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: e56275b82d8a1b0a3a432a233629ac206864f83a

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 074dc717e2c59a75f46c232f5d068130b857d36d
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: e10404c9a10c2ee7f917b59885fa18ede7f0435a

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: aeb34a71e3fd3563bcfadbc47b1b7574c1f9359b
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 3fc96da3fe7b326106f45bec9fcd29d2afa4b3c4

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 5b96721222519db28316c53f605c7f8603965bfc
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 0e5c469765b215fb6c9c7efe7e82d2ab360f0212

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: a2b0d80bb2fe24c63b5611d324b01f653679b0fa
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 48ff5736e8e42a68fceb7e361203ebf0fa1bbeb2

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: 75e241882ae3eba86957aa8ef47647e805b2a28b
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: a573d4fda4232a30130441438b8607ed178df93e

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Merged By
JamieD12

##### Revision
Origin: b0848cac60e6718660d3154778f47593877aaf0e
Pull Request: 2f84df91fe4504da55eac6ecda3bb5a73ddd8fed
Server: 46c827dc9eeae1ddda6382961b987de97833563a

**Review by JamieD1 — APPROVED** [review]
